src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -226,9 +225,8 @@ public static void RangePass()
             Assert.Equal(ushort.MaxValue, JsonSerializer.Parse<ushort>(ushort.MaxValue.ToString()));
             Assert.Equal(ushort.MaxValue, JsonSerializer.Parse<ushort?>(ushort.MaxValue.ToString()));
 
-            // todo: these fail due to double->float conversion
-            //Assert.Equal(float.MaxValue, JsonSerializer.Parse<float>(float.MaxValue.ToString()));
-            //Assert.Equal(float.MaxValue, JsonSerializer.Parse<float?>(float.MaxValue.ToString()));
+            Assert.Equal(float.MaxValue, JsonSerializer.Parse<float>(float.MaxValue.ToString()));
+            Assert.Equal(float.MaxValue, JsonSerializer.Parse<float?>(float.MaxValue.ToString()));
```

Comment:
What is the behavior for parsing `-0` and `-0.0` (you need to check the raw bits, rather than use `Assert.Equal`)?

Reply:
Can you elaborate on why we want to check the raw bits vs. compare `0`, `-0`, and `-0.0` are the same?

Reply:
For IEEE floating-point, `+0` == `-0`, but they have different bit representations. This distinction can be important for some operations (and preserving the sign is required for parsing/formatting) so you need to check the raw bits to ensure you get a zero with the appropriate sign.